### PR TITLE
More Multi or Pipeline check

### PR DESCRIPTION
**Pull request by sazzad16**
First half of solution of #1614

### CODE REVIEW

File: src/main/java/redis/clients/jedis/Jedis.java [hunk 1]
```diff
@@ -2826,6 +2821,18 @@ public void psubscribe(final JedisPubSub jedisPubSub, final String... patterns)
     return params;
   }
 
+  @Override
+  public Object eval(final String script, final int keyCount, final String... params) {
+    checkIsInMultiOrPipeline();
+    client.eval(script, keyCount, params);
+    client.setTimeoutInfinite();
```
Comment (gkorland) [inline src/main/java/redis/clients/jedis/Jedis.java]: don't you think it's better to move `client.setTimeoutInfinite();` into the try block?

Comment (sazzad16) [reply src/main/java/redis/clients/jedis/Jedis.java]: No. `setTimeoutInfinite()` has it's own try-catch. And if there is any exception, it is decided to be broken. There is no gain to `rollbackTimeout()` a broken connection.

**Comment by gkorland** [comment]
@sazzad16 why is it on 3.1.1? should we move it to 3.2.0?

**Comment by sazzad16** [comment]
Because it's **backwards-compatible bug fix**.

**Comment by sazzad16** [comment]
@gkorland 

> why is it on 3.1.1?

Because it's **backwards-compatible bug fix**.

> should we move it to 3.2.0?

Only if you're plan right now is not to have a release 3.1.1.

PS: In that case, we should drop third number while versioning.

**Comment by gkorland** [comment]
I think the third number should be used for "service packs" releasing critical bug fixes for existing version.
The second should be used for "feature version" 
And the first for major versions with breaking backward compatibility.

If you think this issue is critical bug fix on top of 3.1.0 then it should be part of 3.1.1

**Comment by sazzad16** [comment]
I saw may be two issues related this. So I won't call it *"critical"*.

Again, if you've decided that bug fix releases would be only for critical bugs, IMHO, it may be off dropping that third number.

**Pull request merged**